src/main/java/com/aws/greengrass/shadowmanager/model/ShadowState.java

```diff
@@ -33,6 +35,13 @@ public class ShadowState {
     @JsonProperty(SHADOW_DOCUMENT_STATE_REPORTED)
     private JsonNode reported;
 
+    /**
+     * If true, this {@link ShadowState} represents the {"state": null} document, which resets shadow state.
+     */
+    @Setter
+    @JsonIgnore
+    private boolean clear;
```

Comment:
why? We cannot handle nulls without this?

Reply:
we can, although we use shadowDocument.state a bunch of places in the code. thinking was that this would be better than adding null checks everywhere

Reply:
I don't know what it would look like of course, but I think that marking state as `@Nullable` and then handling the null case would make more sense.